ExomeDepth/exomedepth_db.py

```diff
@@ -0,0 +1,231 @@
+        print(session.query(
+            Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+        )
+
+
+def get_flowcelid_bam(bam):
+    with pysam.AlignmentFile(bam, "rb") as workfile:
+        readgroups = []
+        for readgroup in workfile.header['RG']:
+            if readgroup['PU'] not in readgroup:
+                readgroups.append(readgroup['PU'])
+        readgroups = list(set(readgroups))
+        flowcell_id = "_".join(readgroups)
+    return flowcell_id
+
+
+def query_refset_bam(args):
+    Session = connect_database()
+    flowcell_id = get_flowcelid_bam(args.bam)
+    sample_id = determine_sample_id(args.bam)
+    with Session() as session:
+        print(session.query(
+            Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+        )
+
+
+def delete_sample_db(args):
+    Session = connect_database()
+    flowcell_id = get_flowcell_id(args.flowcell_id)
+    with Session() as session:
+        if session.query(Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).all():
+            session.query(Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).delete()
+            session.commit()
+            print("Deleted sample {0} with flowcell {1}.".format(args.sample_id, flowcell_id))
+        else:
+            print("Sample {0} with flowcell {1} not in database.".format(args.sample_id, flowcell_id))
+
+
+def get_flowcell_id(flowcellsarg):
+    flowcells = list(set(flowcellsarg))
+    flowcells.sort()
+    flowcell_id = "_".join(flowcells)
+    return flowcell_id
+
+
+def add_sample_to_db_and_return_refset(args):
+    Session = connect_database()
+    flowcell_id = get_flowcell_id(args.flowcell_id)
+    with Session() as session:
+        if not session.query(Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).all():
+            entry = create_sample(args.sample_id, flowcell_id, args.refset)
+            store_sample(Session, entry)
+        refset = (
+            session.query(Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+        )
+        print(refset)
+        return refset
+
+
+def add_sample_to_db_and_return_refset_bam(args):
+    Session = connect_database()
+    sample_id = determine_sample_id(args.bam)
+    flowcell_id = get_flowcelid_bam(args.bam)
+    with Session() as session:
+        if not session.query(Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).all():
+            entry = create_sample(sample_id, flowcell_id, args.refset)
+            store_sample(Session, entry)
+        refset = (
+            session.query(Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+        )
+        if args.print_refset:
+            print(refset)
+        return refset
```

Comment:
Function is almost identical to `add_sample_to_db_and_return_refset`. Might want to merge overlapping code?

Reply:
add_sample_to_db_and_return_refset has been removed